src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/CompatDifference.cs

```diff
@@ -65,5 +66,22 @@ public CompatDifference(string diagnosticId, string message, DifferenceType type
         /// </summary>
         /// <returns><see cref="string"/> describing the difference.</returns>
         public override string ToString() => $"{DiagnosticId} : {Message}";
+
+        public bool Equals(CompatDifference other) => other != null &&
+                                                      (object.ReferenceEquals(this, other) ||
+                                                      (DiagnosticId.Equals(other.DiagnosticId, StringComparison.InvariantCultureIgnoreCase) &
+                                                      Type.Equals(other.Type) &
+                                                      ReferenceId.Equals(other.ReferenceId, StringComparison.InvariantCultureIgnoreCase)));
+
+        public override bool Equals(object obj) => Equals(obj as CompatDifference);
+
+        public override int GetHashCode()
+        {
+            int hashCode = 1447485498;
```

Comment:
does'nt matter today, but it's good to have comments explaining magic numbers like this. I see this scheme is used in our hashcodes several places in this assembly. A quick web search doesn't throw it up as a typical hashcode parameter..

Reply:
so it could be a terrible one.. can it use HashCode.Combine?

Reply:
It could, in fact that is what we had in other parts of the repo but this being an SDK package it needs to a) target netstandard2.0 (where HashCode.Combine is not inbox) and b) can't have any package dependencies (which would be the way to get to HashCode.Combine in NS2.0). Because of these two reasons, we implement a version of HashCode.Combine here. I did copy this logic from other places in this SDK, but AFAICT it is just a number that was picked and used to generate a hashcode but there isn't any special about it. I think adding comments to explain is a fair point.

Reply:
given the other places don't have comments I wouldn't reset for that.

post 6.0 someone probably should verify this is a reasonable scheme though ...

Reply:
Yeah, might be better to start with a prime.